poh/src/poh_service.rs

```diff
@@ -260,12 +269,13 @@ impl PohService {
                         record.mixin,
                         std::mem::take(&mut record.transactions),
                     );
-                    let (send_res, send_record_result_us) = measure_us!(record.sender.send(res));
-                    debug_assert!(send_res.is_ok(), "Record wasn't sent.");
-
-                    timing.total_send_record_result_us += send_record_result_us;
+                    debug_assert!(res.is_ok(), "Slot {0} wasn't recorded.", record.slot);
                     timing.num_hashes += 1; // note: may have also ticked inside record
-                    if let Ok(new_record) = record_receiver.try_recv() {
+                    if let Some(new_record) = record_receiver.pop() {
+                        if record_receiver.working_bank() != record.slot {
+                            record_receiver.shut_off_producers();
+                            break; // Drop the record.
```

Comment:
we can't drop the `record`, because that means that recording can fail after the insert to ringbuffer. if we're able to insert into the ring buffer, we must **guarantee** that all items in that ringbuffer get processed before we move to the next bank.